### PR TITLE
[#IP-362] [subtask #IP-367] Add pagination to findMessages

### PR DESCRIPTION
**Pull request by AleDore**
#### List of Changes

- Add pagination capabilities to `findMessages`
- Add `paging`utility
- Add new Response type for `id` based pagination

#### Motivation and Context

This PR allows message's pagination due to App' s performance improvements.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/__tests__/paging.test.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        next: anIdBasedModel.id,
+        prev: anIdBasedModel.id
+      }
+    );
+  });
+
+  it("should return an empty page if requested pageSize is 0", async () => {
+    const pageSize = 0 as NonNegativeInteger;
+
+    const iteratorMock = iteratorGenMock([anIdBasedModel, anIdBasedModel, anIdBasedModel, {id: "BBBB" as NonEmptyString}]);
```
Comment (michaeldisaro) [inline src/utils/__tests__/paging.test.ts:38]: Why not having used my previous tests functions? There was a function that generated page of messages with a pending message every N. It was parameterized and allowed me to check also that fillPage correctly fills  a page with messages from different pages preserving the order.

**Review by michaeldisaro — APPROVED** [review]
LGTM. Just 2 point:
* previous paging tests had utilities that created a generic number of messages
* old fillPage function was more functional imho

File: src/utils/__tests__/paging.test.ts (hunk 1 above)
Comment (AleDore) [reply src/utils/__tests__/paging.test.ts:38]: I'd rather to put these kind of tests in integration or even in the unit tests related to the client tag use this pagination utilities :)

**Review by balanza — COMMENTED** [review]
So we need to also add `next` and `prev` to openapi definitions?

File: src/models/message.ts [hunk 2]
```diff
@@ -228,29 +221,80 @@ export class MessageModel extends CosmosdbModel<
...
+   * @param pageSize The requested pageSize
+   * @param maximumMessageId The message ID that can be used to filter next messages (older)
+   * @param minimumMessageId The message ID that can be used to filter previous messages (newest)
    */
   public findMessages(
-    fiscalCode: FiscalCode
-  ): TaskEither<
+    fiscalCode: FiscalCode,
+    pageSize = defaultPageSize,
+    maximumMessageId?: NonEmptyString,
```
[the remark this answers is not in the document]
Comment (michaeldisaro) [reply src/models/message.ts:234]: We thought that since these are "top" and "bottom" limit the maximum and minimun fitted better.

**Review by balanza — APPROVED** [review]
I left some comments, anyway I see nothing wrong so far.

File: src/models/message.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (michaeldisaro) [reply src/models/message.ts:234]: ...and of course I will modify openapi model as needed. I was thinking about maintaining these names.

[the remark this answers is not in the document]
Comment (AleDore) [reply src/models/message.ts:234]: What do you think @michaeldisaro ?

**Comment by AleDore** [comment]
> So we need to also add `next` and `prev` to openapi definitions?

yes, in io-functions-app we have to add these attributes in openapi specs, as far as `hasMoreResults` :)

**Comment by michaeldisaro** [comment]
> > So we need to also add `next` and `prev` to openapi definitions?
> 
> yes, in io-functions-app we have to add these attributes in openapi specs, as far as `hasMoreResults` :)

I already have a branch on those changes. I will modify them according to new specs.

**Pull request merged**